File: snapshots/input/syntax/Main/Interfaces.cs
using System.Diagnostics.CodeAnalysis;

namespace Main;

[SuppressMessage("ReSharper", "all")]
public class Interfaces
{
    public interface IOne
    {
    };

    public interface ITwo
    {
    };

    public interface IThree
    {
    };

    public interface IProperties
    {
        byte Get { get; }
        char Set { set; }
        uint GetSet { get; set; }
        long SetGet { set; get; }
    }

    interface IMethods
    {
        void Nothing();
        int Output();
        void Input(string a);
        int InputOutput(string a);
    };

    interface IEvent
    {
        event EventHandler<int> SomeEvent;
    }

    interface IIndex
    {
        bool this[int index] { get; set; }
    }

    interface IDefault
    {
        void Log(string message)
        {
            Console.WriteLine(message);
        }
    }


    private interface IInherit : IOne, ITwo
    {
    }

    public interface IGetNext<T> where T : IGetNext<T>
    {
        static T operator ++(IGetNext<T> other)
        {
            throw new NotImplementedException();
        }
    }

    private interface ITypeParameter<T1, T2> : ITwo where T1 : IOne where T2 : IThree
    {
    }
}

File: snapshots/input/syntax/VBMain/Interfaces.vb
Imports System.Diagnostics.CodeAnalysis

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class Interfaces
        Interface IOne
        End Interface

        Interface ITwo
        End Interface

        Interface IThree
        End Interface

        Interface IProperties
            ReadOnly Property [Get] As Byte
            WriteOnly Property [Set] As Char
            Property GetSet As UInteger
            Property SetGet As Long
        End Interface

        Interface IMethods
            Sub [Nothing]()
            Function Output() As Integer
            Sub Input(ByVal a As String)
            Function InputOutput(ByVal a As String) As Integer
        End Interface

        Interface IEvent
            Event SomeEvent As EventHandler(Of Integer)
        End Interface

        Interface IIndex
            Default Property Item(ByVal index As Integer) As Boolean
        End Interface

        Private Interface IInherit
            Inherits IOne, ITwo
        End Interface
    End Class
End Namespace
